webapplication/user/views.py

```diff
@@ -24,6 +31,22 @@ def perform_create(self, serializer):
         return user
 
 
+class CustomUserDetailsView(UserDetailsView):
+    def get(self, request, *args, **kwargs):
+        response = super().get(request, *args, **kwargs)
+        component_with_remote_executions = Component.objects.filter(Q(success=True) & Q(geoap_creds_required=True))
+        response.data['server_for_calculation_is_needed'] = component_with_remote_executions.exists()
+        if component_with_remote_executions.exists():
+            response.data['remote_server_available'] = True
+            timeout = urllib3.Timeout(total=5, connect=1.0, read=1.0)
+            http = urllib3.PoolManager(timeout=timeout)
+            try:
+                http.request("GET", os.path.join(settings.DEFAULT_REMOTE_SERVER, "api/request"))
+            except urllib3.exceptions.MaxRetryError:
+                response.data['remote_server_available'] = False
```

Comment:
This must be done in a background process and flag is set. You can use https://waffle.readthedocs.io/en/stable/.

Reply:
fixed